microdnf/context.cpp

```diff
@@ -76,9 +76,47 @@ bool userconfirm(libdnf::ConfigMain & config) {
 
 namespace {
 
-class MicrodnfRepoCB : public libdnf::repo::RepoCallbacks {
+// The "quiete" repo callback class implements callback only for importing repository key, no progress information.
+class MicrodnfQuietRepoCB : public libdnf::repo::RepoCallbacks {
 public:
-    explicit MicrodnfRepoCB(libdnf::ConfigMain & config) : config(&config) {}
+    explicit MicrodnfQuietRepoCB(libdnf::ConfigMain & config) : config(&config) {}
+
+    bool repokey_import(
+        const std::string & id,
+        const std::string & user_id,
+        const std::string & fingerprint,
+        const std::string & url,
+        [[maybe_unused]] long int timestamp) override {
+        auto tmp_id = id.size() > 8 ? id.substr(id.size() - 8) : id;
+        std::cout << "Importing GPG key 0x" << id << ":\n";
+        std::cout << " Userid     : \"" << user_id << "\"\n";
+        std::cout << " Fingerprint: " << fingerprint << "\n";
+        std::cout << " From       : " << url << std::endl;
```

Comment:
Does this printing to `cout` not actually cause issues in quiet mode?

Reply:
It depends on what "quiet" mode means. According to DNF4 documentation, this is not complete silence.
But I made some changes:
The messages is skipped in the "assumeno" case.
I added TODO.
The package list completion code sets "assumeno" set to true.